Tabloid/Repositories/PostRepository.cs

```diff
@@ -50,6 +50,48 @@ public List<Post> GetAllPublishedPosts()
             }
         }
 
+        public List<Post> GetAllPostsByUser(int id)
+        {
+            using (var conn = Connection)
+            {
+                conn.Open();
+                using (var cmd = conn.CreateCommand())
+                {
+                    cmd.CommandText = @"
+                        SELECT p.Id, p.Title, p.Content,
+                               p.ImageLocation AS HeaderImage,
+                               p.CreateDateTime AS PostCreateDateTime,
+                               p.PublishDateTime, p.IsApproved,
+                               p.CategoryId, p.UserProfileId,
+
+                               u.FirstName, u.LastName, u.DisplayName,
+                               u.Email, u.CreateDateTime AS UserCreateDateTime, u.ImageLocation AS AvatarImage,
+                               u.UserTypeId,
+                           
+                               ut.[Name] AS UserTypeName
+                          FROM Post p
```

Comment:
I get that you might need all of this data to satisfy the NewPostFromReader method. I'm wondering whether you might be returning more data than you need back to the requesting client. Maybe you do need it all, just asking you to think about it.

Reply:
Resolved by having two PostFromReader methods, one to return with users, the other to return without them.